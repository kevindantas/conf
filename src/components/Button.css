.Button {
  position: relative;
}

.Button::before {
  content: '';
  width: 60px;
  height: 50px;  
  /*background: 
    linear-gradient(45deg, #12cffe 25%, transparent 25%, transparent 75%, #12cffe 75%, #12cffe), 
    linear-gradient(45deg, #12cffe 25%, transparent 25%, transparent 75%, #12cffe 75%, #12cffe);*/
  background: repeating-linear-gradient(-45deg, transparent, transparent 3px, #61DAFB 3px, #61DAFB 4px);
  position: absolute;
  top: -10px;
  left: -5px;
}


.Button::after {
  content: '';
  width: 60px;
  height: 60px;  
  background: 
    linear-gradient(45deg, #61DAFB 25%, transparent 25%, transparent 75%, #61DAFB 75%, #61DAFB), 
    linear-gradient(45deg, #61DAFB 25%, transparent 25%, transparent 75%, #61DAFB 75%, #61DAFB);
  background-size: 4px 4px;
  position: absolute;
  bottom: -10px;
  right: -10px;
}

