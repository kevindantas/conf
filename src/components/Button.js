import React, { PureComponent } from 'react';
import { css, style } from 'glamor';

import Globals from '../utils/Globals';

const buttonStyle = css({
  borderRadius: 3,
  padding: '14px 80px',
  margin: 16,
  color: Globals.colors.darken,
  fontSize: 28,
  fontWeight: 600,
  fontFamily: 'Rajdhani, sans-serif',
  textAlign: 'center',
  textTransform: 'uppercase',
  letterSpacing: 1,
  cursor: 'pointer',
  border: `1px solid ${Globals.colors.primary}`,
  backgroundColor: Globals.colors.primary,
  transition: Globals.transitions.primary,
});

export default class Button extends PureComponent {

  state = { active: false }

  toggleActive = () => {

    this.setState(prevState => ({
      active: !prevState.active
    }));

  }

  render() {
    const { label, href } = this.props;
    const { active } = this.state;

    return (
      <a className={buttonStyle} href={href}>
        { label }
      </a>
    );
  }
}
