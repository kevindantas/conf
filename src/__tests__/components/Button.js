import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TestUtils from 'react-addons-test-utils';
import Button from '../../components/Button';


describe('<Button />', () => {


  it('should render a label', () => {
    const wrapper = shallow(
      <Button label="MyLabel" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should set `href` prop', () => {
    const link = 'some-link';
    const wrapper = shallow(
      <Button label="MyLabel" href={link} />
    );

    expect(wrapper.find('a').props().href).toEqual(link);
  });


});
