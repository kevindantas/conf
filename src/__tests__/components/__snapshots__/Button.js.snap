exports[`<Button /> should render a label 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className={
        Object {
            "data-css-2dr2o8": "",
          }
    }>
    MyLabel
</button>,
  "nodes": Array [
    <button
      className={
            Object {
                  "data-css-2dr2o8": "",
                }
      }>
      MyLabel
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Button
        label="MyLabel" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "label": "MyLabel",
        },
        "refs": Object {},
        "state": Object {
          "active": false,
        },
        "toggleActive": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className={
                    Object {
                              "data-css-2dr2o8": "",
                            }
          }>
          MyLabel
</button>,
        "_debugID": 2,
        "_renderedOutput": <button
          className={
                    Object {
                              "data-css-2dr2o8": "",
                            }
          }>
          MyLabel
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    label="MyLabel" />,
}
`;
